src/Trace/V2/Gapic/TraceServiceGapicClient.php

```diff
@@ -0,0 +1,544 @@
+     *
+     *     @type \Google\ApiCore\RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @throws \Google\ApiCore\ApiException if the remote call fails
+     * @experimental
+     */
+    public function batchWriteSpans($name, $spans, $optionalArgs = [])
+    {
+        $request = new BatchWriteSpansRequest();
+        $request->setName($name);
+        $request->setSpans($spans);
+
+        $defaultCallSettings = $this->defaultCallSettings['batchWriteSpans'];
+        if (isset($optionalArgs['retrySettings']) && is_array($optionalArgs['retrySettings'])) {
+            $optionalArgs['retrySettings'] = $defaultCallSettings->getRetrySettings()->with(
+                $optionalArgs['retrySettings']
+            );
+        }
+        $mergedSettings = $defaultCallSettings->merge(new CallSettings($optionalArgs));
+        $callable = ApiCallable::createApiCall(
+            $this->traceServiceStub,
+            'BatchWriteSpans',
+            $mergedSettings,
+            $this->descriptors['batchWriteSpans']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Creates a new span.
+     *
+     * Sample code:
+     * ```
+     * try {
+     *     $traceServiceClient = new TraceServiceClient();
+     *     $formattedName = $traceServiceClient->spanName('[PROJECT]', '[TRACE]', '[SPAN]');
+     *     $spanId = '';
+     *     $displayName = new TruncatableString();
+     *     $startTime = new Timestamp();
+     *     $endTime = new Timestamp();
+     *     $response = $traceServiceClient->createSpan($formattedName, $spanId, $displayName, $startTime, $endTime);
+     * } finally {
+     *     $traceServiceClient->close();
+     * }
+     * ```
+     *
+     * @param string $name The resource name of the span in the following format:
+     *
+     *     projects/[PROJECT_ID]/traces/[TRACE_ID]/spans/[SPAN_ID]
+     *
+     * [TRACE_ID] is a unique identifier for a trace within a project;
+     * it is a 32-character hexadecimal encoding of a 16-byte array.
+     *
+     * [SPAN_ID] is a unique identifier for a span within a trace; it
+     * is a 16-character hexadecimal encoding of an 8-byte array.
+     * @param string            $spanId       The [SPAN_ID] portion of the span's resource name.
+     * @param TruncatableString $displayName  A description of the span's operation (up to 128 bytes).
+     *                                        Stackdriver Trace displays the description in the
+     *                                        {% dynamic print site_values.console_name %}.
```

Comment:
Should this placeholder have a value?

Reply:
Heh, that is odd. It seems that it is direct from the proto: https://github.com/googleapis/googleapis/blob/master/google/devtools/cloudtrace/v2/trace.proto#L194

We could manually edit perhaps.

Reply:
A manual edit sounds like a good call

Reply:
Done.